pallets/ddc-payouts/src/lib.rs

```diff
@@ -294,15 +398,41 @@ pub mod pallet {
 
 			let mut updated_billing_report = billing_report.clone();
 			for payee in payees {
-				let _provider = payee.0; // todo: reward provider
-				let share = payee.1;
-				let amount = share.mul(billing_report.total_balance);
-				updated_billing_report.distributed_balance += amount;
+				let node_reward = get_node_reward(
+					&payee.1,
+					&billing_report.total_node_expected_usage,
+					&billing_report.total_node_expected_reward,
+				);
+				let amount = (|| -> Option<u128> {
+					node_reward
+						.transfer
+						.checked_add(node_reward.storage)?
+						.checked_add(node_reward.puts)?
+						.checked_add(node_reward.gets)
+				})()
+				.ok_or(Error::<T>::ArithmeticOverflow)?;
+
+				let node_id = payee.0;
+				let charge: BalanceOf<T> = amount.saturated_into::<BalanceOf<T>>();
+
+				<T as pallet::Config>::Currency::transfer(
+					&updated_billing_report.vault,
```

Comment:
Are we going to use a specific vault per each Billing Report that was generated as a sub-account? I thought we were about to start from a single Pallet's vault account. Did you confirm that sub-accounts are safe?